packages/amplication-server/src/core/build/build.service.ts

```diff
@@ -1331,4 +1331,45 @@ export class BuildService {
       message
     );
   }
+
+  async calcBuildStatus(buildId: string): Promise<EnumBuildStatus> {
+    const build = await this.prisma.build.findUnique({
+      where: {
+        id: buildId,
+      },
+      include: ACTION_INCLUDE,
+    });
+    //if the build status is not unknown, return the status
+    //the function recalculates the status only if the status is unknown
+    if (build.status !== EnumBuildStatus.Unknown)
+      return build.status as EnumBuildStatus;
+
+    if (!build.action?.steps?.length) return EnumBuildStatus.Invalid;
```

Comment:
In this case (Invalid) - don't we want to update in the DB? Is there a chance it will stop being invalid later on?

Reply:
it is a great question. Since we don't really expect this scenario I cant tell when it will happen and because of that I prefer not to update the DB and allow it to be "fixed" some other way

Reply:
Sounds good. In that case, the only additional suggestion I have is writing some warning log (or analytics event?) with the buildId so we can potentially track it later to see if/when it happened (Invalid)